### PR TITLE
Update Index Data Type from Variant to Integer for Excel.WorkSheet.CustomProperties.Item(Index) method

### PR DESCRIPTION
**Pull request by jmhdev**
The CustomProperties collection was not implemented with the CustomProperty.Name indexed.  Entering the name (e.g. as String) of a property throws an error. The documentation is currently inaccurate in stating that you can enter the property name.

### CODE REVIEW

**Comment by Lauragra** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[VBA/Excel-VBA/articles/customproperties-item-property-excel.md](https://github.com/jmhdev/VBA-content/blob/patch-2/VBA/Excel-VBA/articles/customproperties-item-property-excel.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C5%5C3%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201805031632570245-543%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=svBMgbIq2KnfbZQ3vegkcVLUFMM2l1euaYaQDn5Ilkg%3D&st=2018-05-03T16%3A35%3A24Z&se=2018-06-03T16%3A40%3A24Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by Lauragra** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[VBA/Excel-VBA/articles/customproperties-item-property-excel.md](https://github.com/jmhdev/VBA-content/blob/patch-2/VBA/Excel-VBA/articles/customproperties-item-property-excel.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C5%5C3%5Cf8b5b3f0-7390-2e9e-6625-a74fa1ae6f0a%5CPullRequest%5C201805031633074749-543%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=Nx%2FHYRGpbEqv0mcz3GOHsb1DPKBFUqQxmreqsdqMpmk%3D&st=2018-05-03T16%3A42%3A58Z&se=2018-06-03T16%3A47%3A58Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by lindalu-MSFT** [comment]
@jmhdev Hi, thanks so much for your comments and help improving our docs. We really appreciate it! However, I'm not sure this change is correct. I checked with a couple devs who told me that custom properties can be accessed by name. There does seem to be confusion around the name, though, and I did notice a bug report on Stack Overflow. Before I merge yoour PR, please note that in Excel Workbooks (and throughout Office) the expression is CustomDocumentProperties or [DocumentProperties](https://msdn.microsoft.com/en-us/vba/office-shared-vba/articles/documentproperties-members-office), perhaps try that instead?

**Comment by jmhdev** [comment]
Hi @lindalu-MSFT 

I am aware that DocumentProperties/CustomDocumentProperties can be accessed by name.  However, CustomProperties collection objects for worksheets Worksheets cannot.  

Take the following example VBA:

```
Sub SetProp()

Dim sh As Worksheet

Set sh = ThisWorkbook.Worksheets(1)

sh.CustomProperties.Add "Prop1", 1

Debug.Print sh.CustomProperties.Item("Prop1").value

End Sub
```
You will receive a Type Mismatch error:

image

And debugging points to the line trying to access the custom prop via the name:

image

However, if you change "Prop1" to the index number 1, no error is throw:

image

I do believe this change is correct.  CustomDocumentProperties (for a workbook) are indexed by name and number, but the CustomProperties collection on worksheet objects does not have a name-based index.  Please try the example code yourself and see if you are registering the same issue.  

In short:

- Workbook-Level custom properties can be accessed by name and index no.

- Worksheet-Level custom properties can only be accessed by index no.

Per your comment about the stackoverflow bug report, I agree that this is a bug.  I am guessing fixing a VBA implementation is not exactly high on Microsoft's list of things to do with respect to Excel (nor should it be). So, I thought it best to change the docs to reflect reality, even if the reality is that this is 

I was writing a small program that would make use of custom sheet properties.  I was hoping to use the name rather than the index to access the custom props(makes the code more readable), but I will have to make due. 

Thanks for taking the time to review this.  You folks do some awesome work.
